### PR TITLE
fix: Disable copy site for sites that are not atomic

### LINKED ISSUE #72707: Copy Site: Check if the site is atomic before showing Copy Site menu item
### Quick summary

Right now, under `/sites` we don't check if the site is atomic, but only if the site has a copy site feature before showing the "Copy Site" menu item.

A site has an atomic site feature if it has a `Business` plan, which doesn't mean necessarily that it is atomic.

If this is the case, the copy transfer will fail.

### Steps to reproduce

1. Create a site with a `Business` plan. Don't make it atomic.
2. Try to copy the site.
3. Ensure that the copy fails.

### What you expected to happen

The "Copy Site" menu, doesn't appear

### What actually happened

The "Copy Site" menu appears, and the copy transfer fails afterward.

### Impact

One

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

_No response_

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by kozer**
#### Proposed Changes

Don't allow copy site when a site has copy site feature but it's not atomic.

#### Testing Instructions

1. Create a site with a `Business` plan
2. Make sure that it's not atomic.
3. Create another site with a `Business` plan
4. Make sure that the site is atomic.

##### Test copy site from sites dashboard
1. Navigate to `/sites`
2.  Make sure that the newly created site (**not atomic**) _can't_ be copied (no menu item appears)
3. Make sure that the newly created site (**atomic**) _can_ be copied (menu item appears)

##### Test copy site from site tools
1. Navigate to Settings > General > Site Tools (**on your not atomic site**)
2. Make sure that "Copy Site" option _doesn't_ appear.
3. Navigate to Settings > General > Site Tools (**on your atomic site**)
4. Make sure that "Copy Site" option _appears_.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

fixes: https://github.com/Automattic/wp-calypso/issues/72707

### CODE REVIEW

File: client/landing/stepper/hooks/use-site-copy.tsx [hunk 1]
```diff
@@ -29,19 +30,21 @@ function useSafeSiteHasFeature( siteId: number, feature: string ) {
 
 export const useSiteCopy = ( site: SiteExcerptData ) => {
 	const userId = useSelector( ( state ) => getCurrentUserId( state ) );
+	const siteId = site.ID;
```
Comment (wojtekn) [inline client/landing/stepper/hooks/use-site-copy.tsx]: Is there any need to introduce `siteId` constant?

**Review by wojtekn — APPROVED** [review]
Feature-wise looks good and works as expected.

**Comment by sejas** [comment]
@kozer, ~Probably this is not needed since we are checking the `hasCopySiteFeature` `WPCOM_FEATURES_COPY_SITE`.~ 
I've read the issue and I wonder if we can move this validation to the backend.
But this PR seems a good solution too! 👍 

Another option would be asking the user to move the original site to Atomic. inside our stepper.

**Comment by wojtekn** [comment]
> I've read the issue and I wonder if we can move this validation to the backend.

@sejas, the final REST API endpoint performs that validation. Here we should prevent users from entering the flow for not eligible sites, and similarly, we should perform similar validation at the beginning of the flow (1503-gh-Automattic/dotcom-forge).

**Comment by sejas** [comment]
> @sejas, the final REST API endpoint performs that validation. Here we should prevent users from entering the flow for not eligible sites, and similarly, we should perform similar validation at the beginning of the flow ([1503-gh-Automattic/dotcom-forge](https://href.li/?https://github.com/Automattic/dotcom-forge/issues/1503)).

@wojtekn, I was referring to creating a new API endpoint so we can simplify the code in the front end to just one request. We can discuss it as a follow-up or i2.

**Pull request merged**